workflowparsers/quantum_espresso_bands/parser.py

```diff
@@ -0,0 +1,570 @@
+            # 6. (X, X, Y) -> add Y to latest bucket
+            # 7. (X, Y, Y) -> add Y to latest bucket
+            # 8. (X, Y, X) -> add Y and X a new, latest bucket
+            # 9. (X, Y, Z) -> add Y and Z a new, latest bucket
+            if (None not in symmetry_window) and all(
+                [symmetry_window[i] != symmetry_window[i + 1] for i in range(2)]
+            ):
+                segments.append([previous_kpoint])
+            segments[-1].append(kpoint)
+            previous_kpoint = kpoint
+        return segments
+
+    @staticmethod
+    def apply_multiplicity(
+        energies: list[list[float]], mults: list[list[int]]
+    ) -> list[list[float]]:
+        """Apply band multiplicity to energies."""
+        return [
+            [e for e, m in zip(energy, mult) for _ in range(m)]
+            for energy, mult in zip(energies, mults)
+        ]
+
+
+class QuantumEspressoBandsParser:
+    """Parser for Quantum ESPRESSO BANDS calculations."""
+
+    def __init__(self):
+        self.bands_parser = BandsFileParser()
+        self.pwscf_parser = MainfileParser()
+
+    def _find_files(self) -> Optional[str]:
+        """Find BANDS and PWSCF files in the directory."""
+        pw_patt = re.compile(r'Program PWSCF')
+        scf_patt = re.compile(r'Self-consistent Calculation')
+
+        out_files = self.bands_parser.scan_dir_for_files(self.maindir)
+        pwscf_file = ''
+
+        for filepath in out_files:
+            with open(filepath, 'r') as f:
+                f.readline()  # Skip the first line
+                if not pw_patt.search(f.readline()):
+                    continue
+
+                for line in f:
+                    if scf_patt.search(line):
+                        if not pwscf_file:
+                            pwscf_file = filepath
+                        else:
+                            raise ValueError(
+                                f'Multiple PWSCF files with self-consistent calculations found in {self.maindir}. '
+                                f'Cannot determine which one to use for electronic structure information. '
+                                f'Files: {", ".join([pwscf_file, filepath])}'
+                            )
+
+        return pwscf_file if pwscf_file else None
+
+    def parse(self, filepath, archive, logger):
+        """
+        Parse both PWSCF output and BANDS output files.
+
+        Args:
+            filepath: Path to the main output file
+            archive: NOMAD archive where results will be stored
+            logger: Logger for output messages
+        """
+        self.bands_parser.mainfile = filepath
+        self.maindir = os.path.dirname(filepath)
+        self.logger = logger if logger is not None else logging
+        self.archive = archive
+
+        # Find PWSCF files and parse
+        try:
+            if (pwscf_file := self._find_files()) is not None:
+                self.pwscf_parser.mainfile = pwscf_file
```

Comment:
Thsi is already matched and parsed by the quantum espresso parser for PWSCF, you simply have to load the entry if you need to access info from the archive. This means that the BANDS parser should be executed after PWSCF. Set level to 2

Reply:
I'm fine with this approach. I ended up going with a minimalistic parser here, since this is what the other QE workflow parsers did too.
I can apply this, if you're fine with deviating from the old approach.

Reply:
the other qe workflow parsers also do not match and parse pwscf files, look at the parsers init file. there is no deviation afaik, the only thing missing is connecting the entries to pwscf, we should do this in pwscf so we do the child archive generation in only one place

Reply:
Should your or I see to add this to the QE parser then?

Reply:
i can also add it no problem, just do the bands parsing in this pr then.

Reply:
That would be appreciated, as you have a better view on what you want and how to handle child archives.
I will take a look during the review or after then.

Just to be clear: I remove all population of `run`, except for bandstructures, correct? Or do I still copy over e.g. `method` sections?

Reply:
yes just populate the band structure quantities, then the method and system, if you do not see them in the bands mainfile or if they are the same as in pwscf then do not parse them i simply link them with with the pwscf entry.